operators/go/routeagent_controller.go.nolint

```diff
@@ -116,58 +119,70 @@ func (r *ReconcileRouteagent) Reconcile(request reconcile.Request) (reconcile.Re
+		Capabilities:             &corev1.Capabilities{Add: []corev1.Capability{"ALL"}},
+		AllowPrivilegeEscalation: &allow_privilege_escalation,
+		Privileged:               &privileged}
 
-	return &corev1.Pod{
+	routeAgentDaemonSet := &appsv1.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      cr.Name + "-pod",
 			Namespace: cr.Namespace,
+			Name:      "submariner-routeagent",
 			Labels:    labels,
 		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name: "submariner-routeagent",
-					// TODO: Use var here
-					Image: "submariner-route-agent:local",
-					// FIXME: Should be entrypoint script, find/use correct file for routeagent
-					Command:         []string{"submariner-route-agent.sh"},
-					SecurityContext: &security_context_all_cap_allow_escal,
-					VolumeMounts: []corev1.VolumeMount{
-						{Name: "host-slash", MountPath: "/host", ReadOnly: true},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: matchLabels},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: "submariner-routeagent",
+							// TODO: Use var here
+							Image: "submariner-route-agent:local",
+							// FIXME: Should be entrypoint script, find/use correct file for routeagent
+							Command:         []string{"submariner-route-agent.sh"},
+							SecurityContext: &security_context_all_cap_allow_escal,
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "host-slash", MountPath: "/host", ReadOnly: true},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "SUBMARINER_NAMESPACE", Value: cr.Spec.Namespace},
+								{Name: "SUBMARINER_CLUSTERID", Value: cr.Spec.ClusterID},
+								{Name: "SUBMARINER_DEBUG", Value: cr.Spec.Debug},
+								{Name: "SUBMARINER_CLUSTERCIDR", Value: cr.Spec.ClusterCIDR},
+								{Name: "SUBMARINER_SERVICECIDR", Value: cr.Spec.ServiceCIDR},
+							},
+						},
 					},
-					Env: []corev1.EnvVar{
-						{Name: "SUBMARINER_NAMESPACE", Value: cr.Spec.Namespace},
-						{Name: "SUBMARINER_CLUSTERID", Value: cr.Spec.ClusterID},
-						{Name: "SUBMARINER_DEBUG", Value: cr.Spec.Debug},
-						{Name: "SUBMARINER_CLUSTERCIDR", Value: cr.Spec.ClusterCIDR},
-						{Name: "SUBMARINER_SERVICECIDR", Value: cr.Spec.ServiceCIDR},
+					// TODO: Use SA submariner-routeagent or submariner?
+					ServiceAccountName: "submariner-operator",
+					HostNetwork:        true,
+					Volumes: []corev1.Volume{
+						{Name: "host-slash", VolumeSource: corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{Path: "/"}}},
 					},
 				},
 			},
-			// TODO: Use SA submariner-routeagent or submariner?
-			ServiceAccountName: "submariner-operator",
-			HostNetwork:        true,
-			Volumes: []corev1.Volume{
-				{Name: "host-slash", VolumeSource: corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{Path: "/"}}},
-			},
 		},
```

Comment:
nar*: GitHub is the worst..

`*` no action required :)

Reply:
Yeah, not very good diff